nexus/src/app/external_ip.rs

```diff
@@ -114,7 +114,11 @@ impl super::Nexus {
         let pool = match pool {
             Some(pool) => Some(
                 self.ip_pool_lookup(opctx, &pool)?
-                    .lookup_for(authz::Action::Read)
+                    // every authenticated user has CreateChild on IP pools
+                    // because they need to be able to allocate IPs from them.
+                    // The check that the pool is linked to the current silo
+                    // happens inside allocate_floating_ip
+                    .lookup_for(authz::Action::CreateChild)
```

Comment:
Doing a `CreateChild` instead of a `Read` makes sense but `Datastore::allocate_floating_ip` also then does the same lookup? I'd say just drop this lookup and have `allocate_floating_ip` take the `Option<NameOrId>` pool parameter from the floating IP create args directly and only do the lookup there. That's basically what `allocate_probe_ephemeral_ip` does. (`allocate_instance_ephemeral_ip` should do the same in that case as well)

Reply:
Will try it. I noticed it was redundant but wanted to avoid changing much.

Reply:
I'm going to merge this to make sure the fix is in the next dogfood deploy, but I will do that in a followup.